app/models/spree/gateway/amazon.rb

```diff
@@ -135,5 +146,11 @@ def void(response_code, gateway_options)
     def load_amazon_mws(reference)
       @mws ||= AmazonMws.new(reference, gateway: self)
     end
+
+    # A random string of lowercase alphanumeric characters (i.e. "base 36")
+    def random_suffix
```

Comment:
This just gives us more uniqueness with the same numbers of chars without too much trouble.  `.hex` is 0-9a-f (16^10 = 1e12 possibilities) whereas this is 0-9a-z (36^10 = 4e15 possibilities).  Not a huge deal but it's pretty painless and gives 3 orders of magnitude more randomness. :)